### DISCUSSION TITLE
[v5] Custom library for Adonis

### DISCUSSION THREAD

**Discussion by thefrana** [adonisjs/core#2407] [Help] [answered]
Hi,
Adonis was love at first sight. It's really great, so we decided to use it in our company. My task is to create library, that would abstract some unnecessary code, because we are going to have a number of very similar projects. Is there any way to share for example `config/` and `contacts/` folders and other files like exception handlers, middlewares , the `env.ts` and so on?

My idea is to create much simpler dir structure with only a few files, but maintain possibility to customize any behavior by overriding only the files we need. Like I said, I know that most of the projects will be simple ones and really similar. All projects should use same config files.

I think I would need to write a custom provider, but I was unable to do that.

Thanks a lot for any idea

**Comment by thetutlage** [comment, marked as answer]
@lukeeno Your questions is very broad and it will require some initial work on your end before anyone else can jump to providing solutions :)

Maybe start with your library, move everything you like inside that library and then ask small actionable questions one by one. For example: "I have these 3 config files inside my library and I want to inject them to an app"?

**Reply by thefrana** [reply]
You are right. It also comes down to [this task](https://trello.com/c/x8SCwZR9) on roadmap.

Thank you
